src/mzn_bench/mzn_slurm.py

```diff
@@ -194,40 +211,39 @@ async def run_instance(
                 solution = stat_base.copy()
                 solution["status"] = str(result.status)
                 if "time" in result.statistics:
-                    solution["time"] = result.statistics.pop("time").total_seconds()
+                    statistics_time = result.statistics.pop("time")
+                    if type(statistics_time) is timedelta:
+                        solution["time"] = statistics_time.total_seconds() # convert timedelta to seconds (float), since we cannot register and serialize it to YAML
+                    else:
+                        raise TypeError(f"Time statistic was not `datetime.timedelta` type, but was {type(statistics_time)}. Perhaps you are using an older version of minizinc-python")
                 if result.solution is not None:
-                    solution["solution"] = asdict(result.solution)
+                    solution["solution"] = result.solution
                     solution["solution"].pop("_output_item", None)
                     solution["solution"].pop("_checker", None)
-                file.write(ruamel.yaml.dump([solution]))
+                yaml.dump([solution], file)
 
                 statistics.update(result.statistics)
                 statistics["status"] = str(result.status)
                 if result.solution is not None and not is_satisfaction:
-                    statistics["objective"] = result.solution.objective
-
-        total_time = time.perf_counter() - start
-        statistics["time"] = total_time
+                    statistics["objective"] = result.solution["objective"]
     except minizinc.MiniZincError as err:
         statistics["status"] = str(minizinc.result.Status.ERROR)
         statistics["error"] = str(err)
 
+    total_time = time.perf_counter() - start
```

Comment:
One question here is that we always  seem to solution["time"] to minizinc's "time" statistic, but statistic["time"] based on on the mzn-bench `perf_counter` (but not in case of an error, which I changed so you know at what time the error occurred). Perhaps we should use the perf_counter as fallback: `if "time" not in statistics: statistics["time"] = perf_counter() - start`. Otherwise (in the original code), solution["time"] and statistic["time"] might be inconsistent, or the latter might be missing.